### PR TITLE
[extension/googleclientauth] Add GCP auth extension implementation

### PR DESCRIPTION
**Pull request by damemi**
**Description:**
Follow up to boilerplate PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31518
This adds the functions calls to actually implement the extension using our downstream libraries.

It also enables the shutdown tests for this extension (lifecycle tests do not seem possible for us currently)

**Link to tracking Issue:** n/a

**Testing:** downstream unit and integration tests, shutdown tests here

**Documentation:** Here and downstream READMEs

### CODE REVIEW

File: extension/googleclientauthextension/metadata.yaml
Comment (dashpole) [inline extension/googleclientauthextension/metadata.yaml:11]: Should we remove the in development comment?

File: extension/googleclientauthextension/metadata.yaml
Comment (dashpole) [inline extension/googleclientauthextension/metadata.yaml:9]: What else needs to be done to move this from development to alpha stability?

File: extension/googleclientauthextension/metadata.yaml
Comment (damemi) [reply extension/googleclientauthextension/metadata.yaml]: yup done

File: extension/googleclientauthextension/metadata.yaml
Comment (dashpole) [reply extension/googleclientauthextension/metadata.yaml:9]: Ah, I see.  We should keep in in development for this PR, then.

**Review by dashpole — APPROVED** [review]

**Review by dashpole — COMMENTED** [review]
Actually, can you add tests for the config?  Usually, that involves a testdata directory with example configs.  Like this one: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/config_test.go.

It would also be good to check for leaked goroutines: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/googlecloudexporter/package_test.go

**Comment by dashpole** [comment]
Nope, nothing else there.  I just missed them when I looked

**Comment by damemi** [comment]
@dashpole I updated to add config tests, but double check my pointer logic in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32029/commits/aa8288f8217493f3c936b761af5574addc708cea

For this extension we return a pointer default config downstream https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/4caace7de8789f91dc3ddcdbcd328db002c44abb/extension/googleclientauthextension/config.go#L58

But for our exporters, it's not a pointer: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/4caace7de8789f91dc3ddcdbcd328db002c44abb/exporter/collector/config.go#L198

Not sure if this is intentional, but I want to make sure this won't cause a nil pointer in the config. The code is also kind of weird. The alternative is another release downstream to not use a pointer when we call DefaultConfig, then I think the embedded type should just work

**Comment by dashpole** [comment]
Why did you need to add the pointer?

File: extension/googleclientauthextension/factory.go
Comment (dashpole) [inline extension/googleclientauthextension/factory.go:34]: can we just check if it is nil, and dereference the pointer in here?

**Review by dashpole — APPROVED** [review]

**Comment by damemi** [comment]
@dashpole you're right, I could have dereferenced it and think I was misreading the error that I was seeing. I didn't save it, but the issue was with a conflict in the upstream and downstream `Config` types. I think we should refactor the downstream extension code to not rely on the `component.Component` interface if possible and just use a downstream struct. That's closer to what our exporters do. But for now this should work

**Comment by damemi** [comment]
Fixed conflicts, @dashpole this is still ready to merge

**Comment by damemi** [comment]
@djaglowski hi, since you merged the last PR for this component I wanted to see if this was ready to merge? It's been a couple weeks with the ready-to-merge label set and it's tough to keep up with the merge conflicts that keep popping up. Would like to nudge this if possible, thanks!

**Comment by djaglowski** [comment]
@damemi, thanks for the ping. Always feel free to ping me or any other maintainer as soon as an approver marks it ready to merge.

**Pull request merged**
